metadata-io/src/main/java/com/linkedin/metadata/service/UpdateGraphIndicesService.java

```diff
@@ -190,7 +190,10 @@ private void handleDeleteChangeEvent(
               urn.getEntityType(), event.getAspectName()));
     }
 
-    RecordTemplate aspect = event.getRecordTemplate();
+    final RecordTemplate aspect =
```

Comment:
Why can we end up with a `null` aspect? Seems that some of the implementations from `RecordTemplate` can be either `Nullable` and `Nonnull` Just curious if might be just better to return an empty record template that handling a mix of `Nullable` and `Nonnull` that could end up in NPEs.

Reply:
The MCL usually contains both a current aspect and a previous aspect, the previous aspect is null when the aspect is being first created. For a delete, it doesn't make sense that the current aspect would be populated necessarily. I would expect a previous aspect unless we hit a third case which is a delete of the key aspect of an entity which should technically be added separately but is complete redundant with the entity urn.

In any case, we need to respect that the data model defined both current and previous as optional [here](https://github.com/datahub-project/datahub/blob/49b6284ebfa6fae65bf463e0eb3218b9793bb1f2/metadata-models/src/main/pegasus/com/linkedin/mxe/MetadataChangeLog.pdl#L9) and [here](https://github.com/datahub-project/datahub/blob/49b6284ebfa6fae65bf463e0eb3218b9793bb1f2/metadata-models/src/main/pegasus/com/linkedin/mxe/MetadataChangeProposal.pdl#L47). Changing the model is a backwards incompatible change.